l1a_platform/handlers/records.py

```diff
@@ -0,0 +1,198 @@
+    temp_str = ("tcoTemp_str/raw", "tcoTemp_str")
+
+    @property
+    def path(self) -> str:
+        return f"Level1A/HK_tcThermEssential/{self.value[0]}"
+
+
+class AttitudeRecord(BaseRecord):
+    time = ("Time", TIME_KEY)
+    attitude_state = ("afsAttitudeState", "afsAttitudeState")
+    attitude_uncertainty = ("afsAttitudeUncertainty", "afsAttitudeUncertainty")
+    rate_uncertainty = ("afsRateUncertainty", "afsRateUncertainty")
+    spacecraft_rate = ("afsSpacecraftRate", "afsSpacecraftRate")
+    tangent_point = ("afsTangentPoint", "afsTangentPoint")
+
+    @property
+    def path(self) -> str:
+        return f"Level1A/ReconstructedData/PreciseAttitudeEstimation/{self.value[0]}"  # noqa: E501
+
+
+class OrbitRecord(BaseRecord):
+    time = ("Time", TIME_KEY)
+    gnss_state = ("acsGnssStateJ2000", "acsGnssStateJ2000")
+    navigation_uncertainty = (
+        "acsNavigationUncertainty",
+        "acsNavigationUncertainty"
+    )
+
+    @property
+    def path(self) -> str:
+        return f"Level1A/ReconstructedData/PreciseOrbitEstimation/{self.value[0]}"  # noqa: E501
+
+
+class GnssRecord(BaseRecord):
+    time = ("__common__/SCET", TIME_KEY)
+    propagation_time = (
+        "acoOnGnssPropagationTime/raw",
+        "acoOnGnssPropagationTime"
+    )
+    state_vx = ("acoOnGnssStateEcef_vx/raw", "acoOnGnssStateEcef_vx")
+    state_vy = ("acoOnGnssStateEcef_vy/raw", "acoOnGnssStateEcef_vy")
+    state_vz = ("acoOnGnssStateEcef_vz/raw", "acoOnGnssStateEcef_vz")
+    state_x = ("acoOnGnssStateEcef_x/raw", "acoOnGnssStateEcef_x")
+    state_y = ("acoOnGnssStateEcef_y/raw", "acoOnGnssStateEcef_y")
+    state_z = ("acoOnGnssStateEcef_z/raw", "acoOnGnssStateEcef_z")
+    state_time = ("acoOnGnssStateTime/raw", "acoOnGnssStateTime")
+    valid = ("acoOnGnssValid/raw", "acoOnGnssValid")
+
+    @property
+    def path(self) -> str:
+        return f"Level1A/TM_acGnssOps/{self.value[0]}"
+
+
+class HiRateAttitudeRecord(BaseRecord):
+    time = ("__common__/SCET", TIME_KEY)
+    master_return = ("afiStrMasterReturn/raw", "afiStrMasterReturn")
+    res_quaternion_q0 = ("afiStrResQuaternion_q0/raw", "afiStrResQuaternion_q0")
+    res_quaternion_q1 = ("afiStrResQuaternion_q1/raw", "afiStrResQuaternion_q1")
+    res_quaternion_q2 = ("afiStrResQuaternion_q2/raw", "afiStrResQuaternion_q2")
+    res_quaternion_q3 = ("afiStrResQuaternion_q3/raw", "afiStrResQuaternion_q3")
+    cum_sum_angle_x = ("afoRsaCumSumAngleScb_x/raw", "afoRsaCumSumAngleScb_x")
+    cum_sum_angle_y = ("afoRsaCumSumAngleScb_y/raw", "afoRsaCumSumAngleScb_y")
+    cum_sum_angle_z = ("afoRsaCumSumAngleScb_z/raw", "afoRsaCumSumAngleScb_z")
+    validity_count = ("afoRsaValidityCount/raw", "afoRsaValidityCount")
+    mh_obt = ("afoTmMhObt/raw", "afoTmMhObt")
+
+    @property
+    def path(self) -> str:
+        return f"Level1A/TM_afAcsHiRateAttitudeData/{self.value[0]}"
+
+
+def to_utc(gps_datetimes: np.ndarray) -> np.ndarray:
+    return gps_datetimes - np.timedelta64(18, 's')
```

Comment:
I think it would be better for the futuer, if the `np.timedelta` were a named constant, defined up with `GPS_START`.

Reply:
Fixed